clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1568,6 +1571,17 @@ public void handleResponse(AbstractResponse response) {
             Errors error = endTxnResponse.error();
 
             if (error == Errors.NONE) {
+                // For transaction version 5+, the broker includes the producerId and producerEpoch in the EndTxnResponse.
```

Comment:
nit: did you mean EndTxn request version 5+? And maybe we should say transaction version 2 rather than KIP-890 part 2.

Reply:
Oh my yes I did thanks for the catch!